js/bitfinex2.js

```diff
@@ -358,23 +379,68 @@ module.exports = class bitfinex2 extends bitfinex {
         return this.parseTicker (ticker, market);
     }
 
-    parseTrade (trade, market) {
-        let [ id, timestamp, amount, price ] = trade;
-        let side = (amount < 0) ? 'sell' : 'buy';
-        if (amount < 0) {
-            amount = -amount;
+    parseTrade (trade, market = undefined) {
+        if (trade.length > 5) {
```

Comment:
Unfortunately, this line won't transpile to PHP properly, because in JS and Python strings are the same as arrays, but not in PHP. We have to add a type check here:

```JavaScript
if (Array.isArray (trade)) // will get properly transpiled
```

Reply:
You mean:

`if (Array.isArray (trade) && trade.length > 5) {...
`

?

Reply:
Ah, pardon, if it's always an array, then the proper fix is this:

```
const tradeLength = trade.length; 
// ↑ absolutely not intuitive
// but this a hint for the transpiler to treat this as an array, rather than an object
if (tradeLength > 5) {
    // ...
} else {
    // ...
}
```